tests/testthat/test-complete.R

```diff
@@ -13,3 +13,11 @@ test_that("preserves grouping", {
   expect_s3_class(out, "grouped_df")
   expect_equal(dplyr::groups(out), dplyr::groups(df))
 })
+
+test_that('expands empty factors', {
+  emptyfactor <- factor(character(),levels=c('a','b','c'))
```

Comment:
You can just write `factor(levels = c('a', 'b', 'c'))`

(But please use `"` for strings)